### ISSUE TITLE
typing related identifiers

### ISSUE THREAD

**Issue by myrmoteras** [plazi/arcadia-project#106]
@slint we had a discussion regarding related identifiers to show what kind of identifier it is, a publication, image, etc.

e.g. 
Cites
10.5281/zenodo.1182511: image

I understand, that this is possible.

If this is the case, how would we have to do this?

**Comment by slint** [comment]
Yes, it is possible now to select a type for each related identifier:

![image](https://user-images.githubusercontent.com/1814661/66377997-e95ac000-e9b2-11e9-9c09-75401c0307ee.png)

This results in the following at the right sidebar of the published record:

![image](https://user-images.githubusercontent.com/1814661/66378162-35a60000-e9b3-11e9-830b-0881a76cc232.png)


@gsautter for the REST API, this can be included by providing with each related identifier a `resource_type` field like so:

```http
PUT /api/deposit/depositions/12345
{
    "metadata": {
        ...,
        "related_identifiers": [
            {
                "identifier": "10.5281/zenodo.1182511",
                "relation": "cites",
                "resource_type": "publication-article"
            }
        ],
        ...
    }
}
```

The available values are basically the concatenation of the upload types that exist (see [deposit metadata docs](https://developers.zenodo.org/#representation)), i.e.:

- `publication-article`
- `publication-taxonomictreatment`
- ...
- `image-figure`
- `image-plot`
- ...
- `dataset`
- `software`
- ...

This is available on both sandbox and the live instances.

**Comment by gsautter** [comment]
One thing: is it permissible to use only the main type, e.g. `publication`? Would be very helpful for bibliographic citations, as from the citation alone it might not always be possible to determine the exact type of the cited publication.

**Comment by gsautter** [comment]
One more thing: which resource type should we use for materials citations (we're adding them if they have an HTTP URI)? Simply `other`? Or maybe `dataset`?

**Comment by myrmoteras** [comment]
`digital specimen` or `specimen`?

**Comment by gsautter** [comment]
Good question ... nothing in the documentation at https://developers.zenodo.org/#representation seems to really fit, apart from `other` ...

**Comment by gsautter** [comment]
Using `other` for now ... this affects only very few treatments anyway, and we can always change it later.

**Comment by gsautter** [comment]
Done ... the mass update will occur automatically once we add all the treatments.

**Comment by myrmoteras** [comment]
@gsautter and @slint this looks great
https://zenodo.org/record/3476748

![image](https://user-images.githubusercontent.com/4609956/66485418-16899a00-eaa9-11e9-85e0-0c83cc1fa30b.png)
